Add secrets

worksewell opened the pull request:
This adds secret management for teams, clusters, and projects.

Example usage

```
$ gradient secrets set team --name=aws_secret_access_key --value=test
Set team secret 'aws_secret_access_key'

$ gradient secrets list team
+-----------------------+
| Name                  |
+-----------------------+
| aws_secret_access_key |
+-----------------------+

$ gradient secrets delete team --name=aws_secret_access_key
Deleted team secret 'aws_secret_access_key'
```

*NOTE: the APIs to support secrets aren't live yet. We should wait to merge until they are available.*

BartoszCki commented on gradient/api_sdk/clients/secret_client.py:
```diff
@@ -0,0 +1,61 @@
+
+
+class SecretsClient(BaseClient):
+    entity = "secret"
```
This is not required here, I think. This is used for tags. Unless one can add tags to secrets?

worksewell replied:
Nope, will remove.

worksewell commented:
@BartoszCki and/or @kossak: I noticed the entity stuff was moved into a mixin (#280) so I removed the `entity = "secret"` stuff from `SecretsClient` and rebased master (only two changes I made)

The API's for this are now available, so if either of you could re-approve I can get this merged.

Thanks!

P.S. The tests are failing because it looks like the pre-release of numpy uses an f-string in it's setup.py file which fails in the tox python 3.5 tests (python 3.5 isn't supported by numpy anymore, see https://github.com/numpy/numpy/issues/16294).

```
Collecting numpy
 
  Downloading numpy-1.19.0rc1.zip (7.3 MB)
 
  Installing build dependencies: started
 
  Installing build dependencies: finished with status 'done'
 
  Getting requirements to build wheel: started
 
  Getting requirements to build wheel: finished with status 'error'
 
  ERROR: Command errored out with exit status 1:
 
   command: /root/project/.tox/py35-pt46-ptc28/bin/python /root/project/.tox/py35-pt46-ptc28/lib/python3.5/site-packages/pip/_vendor/pep517/_in_process.py get_requires_for_build_wheel /tmp/tmpv1nkape2
 
       cwd: /tmp/pip-install-x16himj_/numpy
 
  Complete output (19 lines):
 
  Traceback (most recent call last):
 
    File "/root/project/.tox/py35-pt46-ptc28/lib/python3.5/site-packages/pip/_vendor/pep517/_in_process.py", line 280, in <module>
 
      main()
 
    File "/root/project/.tox/py35-pt46-ptc28/lib/python3.5/site-packages/pip/_vendor/pep517/_in_process.py", line 263, in main
 
      json_out['return_val'] = hook(**hook_input['kwargs'])
 
    File "/root/project/.tox/py35-pt46-ptc28/lib/python3.5/site-packages/pip/_vendor/pep517/_in_process.py", line 114, in get_requires_for_build_wheel
 
      return hook(config_settings)
 
    File "/tmp/pip-build-env-0boduin5/overlay/lib/python3.5/site-packages/setuptools/build_meta.py", line 148, in get_requires_for_build_wheel
 
      config_settings, requirements=['wheel'])
 
    File "/tmp/pip-build-env-0boduin5/overlay/lib/python3.5/site-packages/setuptools/build_meta.py", line 128, in _get_build_requires
 
      self.run_setup()
 
    File "/tmp/pip-build-env-0boduin5/overlay/lib/python3.5/site-packages/setuptools/build_meta.py", line 250, in run_setup
 
      self).run_setup(setup_script=setup_script)
 
    File "/tmp/pip-build-env-0boduin5/overlay/lib/python3.5/site-packages/setuptools/build_meta.py", line 143, in run_setup
 
      exec(compile(code, __file__, 'exec'), locals())
 
    File "setup.py", line 182
 
      raise ValueError(f'Submodule {p} missing')
 
                                              ^
 
  SyntaxError: invalid syntax
 
  ----------------------------------------
 
ERROR: Command errored out with exit status 1: /root/project/.tox/py35-pt46-ptc28/bin/python /root/project/.tox/py35-pt46-ptc28/lib/python3.5/site-packages/pip/_vendor/pep517/_in_process.py get_requires_for_build_wheel /tmp/tmpv1nkape2 Check the logs for full command output.
 ```

worksewell commented:
@mkulaczkowski Yes, the comment above describes why and they are currently broken on master (see https://github.com/Paperspace/gradient-cli/commits/master).

Do want to remove Python 3.5 support?
